File: python-backend/app/prompts/toilet_paper.py
"""
トイレットペーパー用のプロンプト定義
"""

PROMPT = """
以下のトイレットペーパー商品情報から、正確な情報を抽出してください。

{combined_text}

以下の情報をJSON形式で返してください：  
- roll_count: 実際の物理的なロール数（数値のみ、単位なし）
- length_m: 1ロールあたりの長さ（メートル単位、数値のみ）
- is_double: ダブルかシングルか（true/false/null）

重要な注意事項：
1. 「2倍巻」「3倍巻」「長持ち」商品の場合：
   - 物理的なロール数を使用してください（換算値ではない）
   - 「3倍長持ち 12ロール」→ roll_count: 12（物理的に12ロール）
   - 長さが重要：75mや100mなど、実際の1ロールの長さを正確に取得

2. まとめ買い商品の場合（「×」の後に「パック」が明記されている場合のみ）：
   - 「12ロール×6パック」→ roll_count: 72（パック数を掛ける）
   - ただし「80m×8ロール」のような場合は長さ×ロール数の表記なので、roll_count: 8

3. 長さの解釈：
   - 「50m x 12ロール」→ 1ロール50m、総数12ロール
   - 長さの単位はメートルに統一
   - mm（ミリメートル）は幅を示すので長さとして使用しない

4. ダブル/シングル判定：
   - 「2枚重ね」「ダブル」→ is_double: true
   - 「シングル」「1枚」→ is_double: false
   - 不明な場合→ is_double: null

JSONのみを返してください。説明は不要です。
"""

FIELDS = {
    'roll_count': None,
    'length_m': None,
    'is_double': None
}

def post_process(extracted_info):
    """トイレットペーパー固有の後処理"""
    # 総長さの計算
    if extracted_info['roll_count'] and extracted_info['length_m']:
        extracted_info['total_length_m'] = extracted_info['roll_count'] * extracted_info['length_m']
    else:
        extracted_info['total_length_m'] = None
    return extracted_info

File: lib/products/toilet-paper.ts
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!;
const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!;

const supabase = createClient(supabaseUrl, supabaseAnonKey);

export interface ToiletPaperProduct {
  asin: string;
  title: string;
  description?: string;
  brand?: string;
  image_url?: string;
  price?: number;
  price_regular?: number;
  discount_percent?: number;
  on_sale: boolean;
  review_avg?: number;
  review_count?: number;
  roll_count?: number;
  length_m?: number;
  total_length_m?: number;
  price_per_roll?: number;
  price_per_m?: number;
  is_double?: boolean;
  last_fetched_at?: string;
}

export async function getToiletPaperProducts(): Promise<ToiletPaperProduct[]> {
  const { data, error } = await supabase
    .from('toilet_paper_products')
    .select('*')
    .not('price', 'is', null)  // 在庫切れを除外
    .order('price_per_m', { ascending: true, nullsFirst: false });

  if (error) {
    console.error('Error fetching toilet paper products:', error);
    return [];
  }

  return data || [];
}

export async function getToiletPaperProduct(asin: string): Promise<ToiletPaperProduct | null> {
  const { data, error } = await supabase
    .from('toilet_paper_products')
    .select('*')
    .eq('asin', asin)
    .single();

  if (error) {
    console.error('Error fetching toilet paper product:', error);
    return null;
  }

  return data;
}